### ISSUE TITLE
errors in running find_de_neighborhoods

### ISSUE THREAD

**Issue by JLiLab** [const-ae/lemur#6]
Thank you for developing the method. When I tested the example data, it ran fine. However, I got the following error message when I tried our own data from an Seurat object: 

sce = as.SingleCellExperiment(seu, assay = "RNA")
fit <- lemur(sce, design = ~ batch+group, n_embedding = 15, verbose = FALSE)
fit <- test_de(fit, contrast = cond(group = "cKO") - cond(group = "CTRL"))

nei <- find_de_neighborhoods(fit, independent_matrix = counts(sce), goup_by = vars(batch,group))
Find optimal neighborhood using zscore.
Validate neighborhoods using independent counts data.
Form pseudobulk (summing counts)
Error in value[[3L]](cond) : 
  Error in doTryCatch(return(expr), name, parentenv, handler): The 'group_by' argument must be wrapped using 'vars()'

The 'group_by' argument must be wrapped using 'vars()'

Any idea what is the issue?

By the way, I can ran the following:
nei <- find_de_neighborhoods(fit, group_by = vars(group))
head(nei)

    name selection      indices n_cells sel_statistic
1   Cdh9      TRUE 5, 7, 15....    4928     -34.34182
2    Ttr      TRUE 1, 2, 3,....   15112     174.56091
3   Gpc5      TRUE 1, 2, 5,....    4213     159.93325
4  Htr2c      TRUE 1, 5, 7,....    3336      49.64083
5 Lrrc4c      TRUE 2, 5, 6,....   13294     112.90822
6 Brinp3      TRUE 13, 68, ....     848     -95.11558

The results have no p values.

Thank you, 

James

**Comment by const-ae** [comment]
Hi James,

> `nei <- find_de_neighborhoods(fit, independent_matrix = counts(sce), goup_by = vars(batch,group))`

I think the error is caused by a typo. It should be `group_by` and not `goup_by`.

----------------

Also note that I recently refactored the `find_de_neighborhoods` function ([NEWS](https://github.com/const-ae/lemur/blob/master/NEWS.md#v0019)). The new version automatically sets some of the data aside as a test set which is not used to infer the parameters of the LEMUR model, but only for the differential expression test. As a consequence, I renamed some of the arguments in `find_de_neighborhoods`.

Best,
Constantin
